README.md
#Notice

Notice is a jQuery plugin for simple notifications

##Usage

```javascript
//info
$.notice('message');

//success
$.notice('success!', { level: 'success' });

//error
$.notice('error', { level: 'error' });
```

##Options

```javascript
$.notice.defaults = {
	//container to append notice to
	container: 'body',
	//template (shouldn't be changed)
	template: '<div class="notice"><span class="notice-text"></span><div class="notice-close"></div></div>',
	//height of notice
	height: 30,
	//timeout before auto closing, set to false to disable auto close
	timeout: 5000,
	//level (info, success, error)
	level: 'info',
	//top offset for notice
	offsetTop: 0,
	//left offset for notice
	offsetLeft: 0,
	//width of notice
	width: '100%'
};
```

##Installation
CSS provides some default styles, it is optional.

- Development: [JS](https://raw.github.com/jgallen23/notice/master/dist/notice.js) | [CSS](https://raw.github.com/jgallen23/notice/master/dist/notice.css)
- Production: [JS](https://raw.github.com/jgallen23/notice/master/dist/notice.min.js) | [CSS](https://raw.github.com/jgallen23/notice/master/dist/notice.min.css)

##Building

```
npm install
./node_modules/.bin/grunt
```

##Development

```
./node_modules/.bin/grunt dev
```


HISTORY.md

0.3.2 / 2013-08-07 
==================

  * updated build process
  * be able to pass in zIndex as option
  * watch styl files
  * updated to latest grunt, added bower.json

0.3.1 / 2012-10-30 
==================

  * removed rgba from theme

0.3.0 / 2012-10-30 
==================

  * automatically fit to it's container
  * built css with banner
  * added banner to css files
  * updated readme
  * Merge branch 'feature/grunt'
  * built
  * updated readme
  * removed unnecessary files
  * added grunt and package.json

0.2.0 / 2012-10-24 
==================

  * updated readme
  * updated example
  * added themes
  * commented options

0.1.0 / 2012-10-23 
==================

  * refactored a bit, added example, makefile
  * changed height to integer
  * offsetTop and ('close') method
  * added default level, option do disable timeout and default styles
  * initial commit


grunt.js
module.exports = function(grunt) {
  grunt.loadNpmTasks('grunt-contrib-stylus');
  grunt.initConfig({
    info: '<json:package.json>',
    meta: {
      banner: '/*!\n'+
              ' * <%= info.name %> - <%= info.description %>\n'+
              ' * v<%= info.version %>\n'+
              ' * <%= info.url %>\n'+
              ' * copyright <%= info.company %> <%= grunt.template.today("yyyy") %>\n'+
              ' * <%= info.license %> License\n'+
              '*/'
    },
    lint: {
      all: 'lib/notice.js'
    },
    stylus: {
      flatten: {
        files: {
          'dist/*.css': 'lib/*.styl'
        }
      },
      compress: {
        options: {
          compress: true
        },
        files: {
          'dist/notice.min.css': 'lib/*.styl'
        }
      }
    },
    concat: {
      dist: {
        src: ['<banner>', 'lib/notice.js'],
        dest: 'dist/notice.js'
      },
      css: {
        src: ['<banner>', 'dist/notice.css'],
        dest: 'dist/notice.css'
      },
      cssmin: {
        src: ['<banner>', 'dist/notice.min.css'],
        dest: 'dist/notice.min.css'
      }
    },
    min: {
      dist: {
        src: ['<banner>', 'dist/notice.js'],
        dest: 'dist/notice.min.js'
      }
    },
    watch: {
      css: {
        files: 'lib/notice.styl',
        tasks: 'stylus'
      },
      js: {
        files: 'lib/notice.js',
        tasks: 'concat min' 
      }
    },
    server:{
      port: 8000,
      base: '.'
    }
  });
  grunt.registerTask('default', 'lint stylus concat min');
  grunt.registerTask('dev', 'server watch');
}


Gruntfile.js
module.exports = function(grunt) {
  grunt.initConfig({
    info: grunt.file.readJSON('bower.json'),
    meta: {
      banner: '/*!\n'+
              ' * <%= info.name %> - <%= info.description %>\n'+
              ' * v<%= info.version %>\n'+
              ' * <%= info.homepage %>\n'+
              ' * copyright <%= info.copyright %> <%= grunt.template.today("yyyy") %>\n'+
              ' * <%= info.license %> License\n'+
              '*/\n'
    },
    jshint: {
      main: [
        'Gruntfile.js',
        'bower.json',
        'lib/**/*.js',
        'test/*.js'
      ]
    },
    bower: {
      main: {
        dest: 'dist/_bower.js',
        exclude: [
          'assert',
          'jquery'
        ]
      }
    },
    stylus: {
      flatten: {
        files: {
          'dist/notice.css': 'lib/*.styl'
        }
      }
    },
    concat: {
      options: {
        banner: '<%= meta.banner %>'
      },
      css: {
        src: ['<banner>', 'dist/notice.css'],
        dest: 'dist/notice.css'
      },
      dist: {
        src: 'lib/notice.js',
        dest: 'dist/notice.js'
      }
    },
    uglify: {
      options: {
        banner: '<%= meta.banner %>'
      },
      dist: {
        src: 'dist/notice.js',
        dest: 'dist/notice.min.js'
      }
    },
    clean: {
      bower: [
        'dist/_bower.js'
      ],
      dist: [
        'dist'
      ]
    },
    watch: {
      scripts: {
        files: '<%= jshint.main %>',
        tasks: 'scripts',
        options: {
          livereload: true
        }
      },
      example: {
        files: [
          'example/*'
        ],
        options: {
          livereload: true
        }
      },
      grunt: {
        files: [
          'Gruntfile.js',
          'test/index.html'
        ],
        tasks: 'default'
      }
    },
    mocha: {
      all: {
        src: 'test/index.html',
        options: {
          run: true,
          growl: true
        }
      }
    },
    plato: {
      main: {
        files: {
          'reports': ['lib/*.js']
        }
      }
    },
    connect: {
      server:{
        options: {
          hostname: '*'
        }
      },
      plato: {
        port: 8000,
        base: 'reports',
        options: {
          keepalive: true
        }
      }
    },
    bytesize: {
      scripts: {
        src: [
          'dist/*'
        ]
      }
    }
  });
  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks('grunt-contrib-concat');
  grunt.loadNpmTasks('grunt-contrib-watch');
  grunt.loadNpmTasks('grunt-contrib-uglify');
  grunt.loadNpmTasks('grunt-contrib-connect');
  grunt.loadNpmTasks('grunt-contrib-clean');
  grunt.loadNpmTasks('grunt-concat-bower');
  grunt.loadNpmTasks('grunt-shell');
  grunt.loadNpmTasks('grunt-bytesize');
  grunt.loadNpmTasks('grunt-mocha');
  grunt.loadNpmTasks('grunt-contrib-stylus');
  grunt.loadNpmTasks('grunt-contrib-cssmin');
  grunt.loadNpmTasks('grunt-plato');
  grunt.registerTask('scripts', ['jshint', 'bower', 'concat', 'uglify', 'clean:bower', 'bytesize']);
  grunt.registerTask('styles', ['stylus', 'concat:css']);
  grunt.registerTask('default', ['styles', 'scripts']);
  grunt.registerTask('dev', ['default', 'connect:server', 'watch']);
  grunt.registerTask('reports', ['plato', 'connect:plato']);
};


lib/notice.js
(function($) {
  var timeout;
  $.notice = function(message, options) {

    var opts = $.extend({}, $.notice.defaults, options);
    var el;

    var close = function() {
      if (timeout) {
        clearTimeout(timeout);
      }

      $('.notice').animate({ height: 0 }, {
        complete: function() {
          $(this).remove();
        }
      });
    };

    //close any open messages
    close();

    if (message == 'close') {
      return;
    }

    var container = $(opts.container);
    var containerOffset = container.offset();
    el = $(opts.template)
      .css({
        zIndex: opts.zIndex,
        overflow: 'hidden',
        position: 'fixed',
        top: opts.offsetTop || containerOffset.top,
        left: containerOffset.left,
        height: 0,
        width: container.width(),
        margin: '0 auto',
        'line-height': opts.height+'px',
        'text-align': 'center'
      })
      .addClass((opts.level)?'notice-'+opts.level:false)
      .find('.notice-text')
        .html(message)
        .end()
      .find('.notice-close')
        .on('click', function() {
          close();
        })
        .end()
      .appendTo(container)
      .animate({ height: opts.height+'px'});

    if (typeof opts.timeout === 'number') {
      timeout = setTimeout(function() {
        close();
      }, opts.timeout);
    }

    return el;
  };

  $.notice.defaults = {
    //container to append notice to
    container: 'body',
    //template (shouldn't be changed)
    template: '<div class="notice"><span class="notice-text"></span><div class="notice-close"></div></div>',
    //height of notice
    height: 30,
    //timeout before auto closing, set to false to disable auto close
    timeout: 5000,
    //level (info, success, error)
    level: 'info',
    //top offset for notice
    offsetTop: 0,
    zIndex: 1000

  };
})(window.jQuery || window.Zepto);
